Update Dockerfile to make container build process more comprehensive

johndowns opened the pull request:
This PR makes two changes to the Dockerfile for this container image:

1. It downloads the SHIR MSI file automatically as part of the container build process.

1. It uses the `ENTRYPOINT` instruction instead of `CMD`. This ensures that the SHIR always starts when the container starts, including on platforms like Azure App Service that don't automatically run `CMD` instructions.

xumou-ms commented:
I think we shouldn't blandly download the latest SHIR. It will break the current use case describe in our public doc. Because we ask user to put the downloaded SHIR to SHIR folder.

Maybe it's better to add a logic in setup.ps1: 

If we can't find a file match the pattern IntegrationRuntime.*.msi, we will download the latest SHIR. Otherwise, we directly use the file.

johndowns commented:
@xumou-ms I have updated the Dockerfile to remove the download step, and added it to the PowerShell script with a condition to check whether the file already exists.
